src/aai/aai.js

```diff
@@ -21,7 +21,24 @@
       var targetUrl = '';
       var opts = jQuery.extend({}, this.defaults, options),
           defaultCallback = function(e) {
-            targetUrl = opts.target + '?redirectUrl=' + window.location.href;
+            targetUrl = opts.target + '?redirectUrl=';
+            var redirectUrl = '';
+
+            // Redirection could be initiated from the login page; in that case,
+            // we need to retrieve the redirect URL from the URL parameters
+            var urlParams = new URLSearchParams(window.location.href.split('?')[1]);
```

Comment:
can there be multiple `?`?

Reply:
Yes, https://stackoverflow.com/questions/2924160/is-it-valid-to-have-more-than-one-question-mark-in-a-url